aiida/post-init.sh

```diff
@@ -60,5 +61,5 @@ verdi profile setup core.sqlite_dos \
 {% endif %}
 
 verdi config set warnings.rabbitmq_version False
-echo "consumer_timeout = 36000000000" > /opt/conda/pkgs/rabbitmq-server-3.12.12-ha770c72_0/etc/rabbitmq/rabbitmq.conf
+echo "consumer_timeout = 36000000000" > /opt/conda/pkgs/rabbitmq-server-3.13.1-ha770c72_0/etc/rabbitmq/rabbitmq.conf
```

Comment:
This will break the next time `aiida-core` upgrades `rabbitmq` and will cause the container to crash. I believe you can do the same by setting an environment variable? See [here](https://stackoverflow.com/a/73928874/1065823). I suppose you could set that in the `Dockerfile`.

Reply:
As both, the versions of `aiida-core` and `aiida-core.services` are pinned in the `Dockerfile`, it should not break, though? Anyway, I agree that this is something that can already be included during the build. Thanks for sending the link, I wasn't aware of that.

Btw, just out of curiosity, is there a recommended way to build the image? My current approach is to modify the `Dockerfile`, build it locally, and then push it to the registry, i.e.:
```shell
docker buildx build -t registry.renkulab.io/julian.geiger/aiida-renku-dev:latest ./aiida
docker push registry.renkulab.io/julian.geiger/aiida-renku-dev:latest
```
But for that to work, I'm always modifying this part:

```dockerfile
# RENKU_VERSION determines the version of the renku CLI
# that will be used in this image. To find the latest version,
# visit https://pypi.org/project/renku/#history.
ARG RENKU_VERSION={{ __renku_version__ | default("2.9.1") }}

# For local build
# ARG RENKU_VERSION="2.9.1"
```
to disable the `jinja` variable[^*], build the image, and enable it again before I commit. Building via the GitLab repo CI also seems a bit suboptimal, as I want the change to be applied to this repo, not the one I mirror on GitLab for the container registry. Thanks!

[^*]: Which gets filled in when the project is created, if I understand correctly?

Reply:
right, `aiida-core` is pinned, but there is no knowledge in the files here about which version of rabbitmq it bundles. So if you forget that this needs to be updated, it will fail. 

Regarding the image build - it's true that it isn't easy to build from the templated Dockerfile... I'm wondering if we could build the image in an action and push it, like we do [here](https://github.com/SwissDataScienceCenter/renkulab-docker/blob/main/.github/workflows/build-and-push-to-docker.yml) for our base images?

Reply:
`aiida-core.services` is pinned via `conda` so shouldn't that specify also the RMQ version (the `aiida-core` services are basically RMQ and PostgreSQL)? I'm not familiar with `conda` packaging so I might be wrong, though. In any case, I agree that modifying the config file in the hardcoded path is far from a desirable solution. I have changed it to use the ENV variable in the `Dockerfile`. Will give it a spin to see if it works and update here. Then we should hopefully be ready to merge.

Alright, at least I know I haven't been doing it in a wrong way. Yeah, an action would be preferrable I guess. For now, I think it's fine to leave it, but one could extend the [documentation](https://renku.readthedocs.io/en/0.51.0/reference/templates.html#pin-a-docker-image) in addition to the [Discourse post](https://renku.discourse.group/t/how-to-pin-a-docker-image/240). For me it was a bit confusing that I'm editing a repo from GitHub, but then the Docker image is hosted on the container registry of another project on GitLab via its CI/CD pipeline. So maybe just adding this option of local bulid and push to the container registry (once the project actually exists on GitLab) via, e.g. the two commands I posted used could be helpful :)